tools/local-snapshot-runner-app/src/exportFile.ts

```diff
@@ -0,0 +1,72 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Loader } from "@fluidframework/container-loader";
+import { createLocalOdspDocumentServiceFactory } from "@fluidframework/odsp-driver";
+import { ChildLogger, PerformanceEvent } from "@fluidframework/telemetry-utils";
+import * as fs from "fs";
+import FileLogger from "./logger/FileLogger";
+import { getArgsValidationError } from "./getArgsValidationError";
+import { isCodeLoaderBundle } from "./codeLoaderBundle";
+import { FakeUrlResolver } from "./fakeUrlResolver";
+import path from "path";
+import { ITelemetryBaseLogger } from "@fluidframework/common-definitions";
+
+export async function exportFile(
+    codeLoader: string,
+    inputFile: string,
+    outputFolder: string,
+    scenario: string,
+    telemetryFile: string,
+    loggerOverride?: ITelemetryBaseLogger,
+) {
+    if (fs.existsSync(telemetryFile)) {
+        console.error("Telemetry file already exists. " + telemetryFile);
+        throw new Error("Telemetry file already exists.");
+    }
+
+    const logger = ChildLogger.create(loggerOverride ?? new FileLogger(telemetryFile), "LocalSnapshotRunnerApp");
```

Comment:
not sure about possible override (not sure I see a scenario), but for file logger we want to add a property (functor) that outputs current time. With other workflows, we rely on external implementation of telemetry interface to add a bunch of common properties include event sequence, event time, some correlation IDs, machine info, etc. In general, most of that should be not responsibility of this code (external code can do it), but not for time - once the event is written and time is not captured, there is no way to fill that data.